base/abstractarray.jl

```diff
@@ -1146,7 +1146,7 @@ parts can specialize this method to return the concatenation of the `dataids` of
 their component parts.  A typical definition for an array that wraps a parent is
 `Base.dataids(C::CustomArray) = dataids(C.parent)`.
 """
-dataids(A::AbstractArray) = (UInt(objectid(A)),)
```

Comment:
What about immutable array types?

Reply:
Aha, yes, thank you.  This isn't the complete solution.

Reply:
If those aren't very important I imagine you can either fallback to `objectid` or `0`?

Reply:
It would probably be easier to add a codegen optimization for `objectid`. For mutable types, it's just a simple bitmix call.